sdks/python/apache_beam/tools/distribution_counter_microbenchmark.py

```diff
@@ -0,0 +1,56 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""A microbenchmark for measuring DistributionAccumulator performance
+
+This runs a sequence of distribution.update for random input value to calculate
+average update time per input.
+A typical update operation should run into 0.6 microseconds
+
+Run as
+  python -m apache_beam.tools.distribution_counter_microbenchmark
+"""
+
+from __future__ import print_function
+
+import random
+import sys
+import time
+
+from apache_beam.tools import utils
+from apache_beam.transforms.cy_combiners import DistributionAccumulator
+
+
+def run_benchmark(num_runs=100, num_input=10000,
+                  low_int=0, high_int=sys.maxint):
+  total_time = 0
+  inputs = random.sample(xrange(low_int, high_int), num_input)
```

Comment:
There is no `xrange` in python 3 and we recently removed it from other places to be compatible. How about just using range with a not so large number? Or if you want to full distribution, perhaps really do it in a loop using randint instead.